client/ayon_traypublisher/plugins/create/create_editorial_advanced.py

```diff
@@ -0,0 +1,1341 @@
+            # add collected files to list
+            files_ = [
+                file for file in files
+                if file.startswith(head)
+                if file.endswith(tail)
+            ]
+            extension = os.path.splitext(files_[0])[1]
+            suffix = differences[files_[0]]
+
+            if match:
+                # remove product name from suffix
+                self.log.warning(f"match[1]: {match[1]}")
+                self.log.warning(f"match[0]: {match[0]}")
+                suffix = suffix.replace(match[1] or match[0], "")
+
+            content_type = "other"
+            if (
+                extension in VIDEO_EXTENSIONS
+                or extension in IMAGE_EXTENSIONS
+            ):
+                content_type = "single"
+
+            # check if file is thumbnail
+            if "thumb" in reminder:
+                content_type = "thumbnail"
+
+            product_data = deepcopy(product_data_base)
+            product_data.update({
+                "type": content_type,
+                "suffix": suffix,
+                "files": files_,
+            })
+
+            if strict and match:
+                # Extract matched pattern and handle special cases with dots
+                # like name.thumbnail.jpg matches
+                matched_pattern = match[1]
+
+                product_data["product_name"] = matched_pattern
+
+            self.log.warning(f"product_data: {pformat(product_data)}")
+
+            collecting_items.append(product_data)
+
+    def _restore_otio_source_range(self, otio_clip):
+        """Infusing source range.
+
+        Otio clip is missing proper source clip range so
+        here we add them from from parent timeline frame range.
+
+        Args:
+            otio_clip (otio.Clip): otio clip object
+        """
+        otio_clip.source_range = otio_clip.range_in_parent()
+
+    def _make_product_instance(
+        self,
+        product_preset,
+        base_instance_data,
+        parenting_data,
+        clip_content_items,
+        media_folder_path,
+    ):
+        """Creating product instances
+
+        Args:
+            product_preset (dict): product preset data
+            base_instance_data (dict): base instance data
+            parenting_data (dict): parenting data
+            clip_content_items (list[dict]): list of clip content items
+            media_folder_path (str): media folder path
+        """
+        pres_product_type = product_preset["product_type"]
+        pres_product_name = product_preset["product_name"]
+        pres_versioning = product_preset["versioning_type"]
+        pres_representations = product_preset["representations"]
+
+        # Dictionary to group files by product name
+        grouped_representations = {}
+
+        # First pass: group matching files by product name and representation
+        for item in clip_content_items:
+            item_type = item["type"]
+            if pres_product_name not in item["preset_name"]:
```

Comment:
I guess this should be explicit string comparison?
```suggestion
            if pres_product_name != item["preset_name"]:
```